xbmc/platform/linux/input/LIRC.cpp

```diff
@@ -18,389 +18,120 @@
  *
  */
 
-#include "threads/SystemClock.h"
-#include <sys/time.h>
-#include <sys/types.h>
-#include <sys/socket.h>
-#include <sys/un.h>
-#include <limits.h>
-#include <unistd.h>
 #include "LIRC.h"
-#ifdef HAVE_INOTIFY
-#include <sys/inotify.h>
-#endif
+#include "Application.h"
```

Comment:
No introduction of globals in the input code

Reply:
Capp is not a real global, it is the container for all other objects. It is CApp that owns the main thread and messages from input or windowing needs to be sent to Capp.
Originally ApplicationMessenger was in charge to accept messages for the main thread but it has been abusued for a ot of other things.
Capp needs an inbound port for messages that solely serves this purpose.

Reply:
Then lets add an interface for inbound messages instead of referencing g_application directly.

Reply:
If you want this is the scope of this PR, fixing Lirc will almost affect all platforms. g_application.OnEvent is used by windowing.

Reply:
or we can invert dependency. i.e. a creator depends on an input and the creator depends on CApp. CLirc produces events, the creator subscribe to receive these events, the creator forwards events from the input to the CApp. this IoC makes weak dependency between an input and CApp and only creator bind them